.github/workflows/otel-fork-replace.yml

```diff
@@ -38,38 +38,39 @@ jobs:
-          go mod tidy
-          go mod edit -replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs=github.com/amazon-contributing/opentelemetry-collector-contrib/internal/aws/cwlogs@${{ steps.get-latest-commit.outputs.sha }}
-          go mod tidy
-          go mod edit -replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awscloudwatchlogsexporter=github.com/amazon-contributing/opentelemetry-collector-contrib/exporter/awscloudwatchlogsexporter@${{ steps.get-latest-commit.outputs.sha }}
-          go mod tidy
-          go mod edit -replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsxrayreceiver=github.com/amazon-contributing/opentelemetry-collector-contrib/receiver/awsxrayreceiver@${{ steps.get-latest-commit.outputs.sha }}
-          go mod tidy
-          go mod edit -replace github.com/amazon-contributing/opentelemetry-collector-contrib/override/aws=github.com/amazon-contributing/opentelemetry-collector-contrib/override/aws@${{ steps.get-latest-commit.outputs.sha }}
-          go mod tidy
-          go mod edit -replace github.com/amazon-contributing/opentelemetry-collector-contrib/pkg/stanza=github.com/amazon-contributing/opentelemetry-collector-contrib/pkg/stanza@${{ steps.get-latest-commit.outputs.sha }}
-          go mod tidy
+          packages=(
+            exporter/awscloudwatchlogsexporter
+            exporter/awsemfexporter
+            exporter/awsxrayexporter
+            extension/awsmiddleware
+            extension/awsproxy
+            internal/aws/awsutil
+            internal/aws/containerinsight
+            internal/aws/cwlogs
+            internal/aws/k8s
+            internal/aws/proxy
+            internal/aws/xray
+            internal/kubelet
+            pkg/translator/prometheus
+            processor/resourcedetectionprocessor
+            receiver/awscontainerinsightreceiver
+            receiver/awsxrayreceiver
+            receiver/prometheusreceiver
+          )
+          for p in ${packages[@]}; do
+            go mod edit -replace github.com/open-telemetry/opentelemetry-collector-contrib/$p=github.com/amazon-contributing/opentelemetry-collector-contrib/$p@${{ steps.get-latest-commit.outputs.sha }}
+            go mod tidy
+          done
+          
+          amazonContributingPackages=(
+            override/aws
+            pkg/stanza  
+          )
+          for p in ${amazonContributingPackages[@]}; do
+            go mod edit -replace github.com/amazon-contributing/opentelemetry-collector-contrib/$p=github.com/amazon-contributing/opentelemetry-collector-contrib/$p@${{ steps.get-latest-commit.outputs.sha }}
+            go mod tidy
+          done
+          
           git commit -am "Update OTel fork components to https://github.com/amazon-contributing/opentelemetry-collector-contrib/commit/${{ steps.get-latest-commit.outputs.sha }}"
```

Comment:
nit:
```suggestion
          git commit -am "Update amazon-contributing/opentelemetry-collector-contrib dependencies to @${{ steps.get-latest-commit.outputs.sha }}"
```

Reply:
Can you also remove
https://github.com/aws/amazon-cloudwatch-agent/blob/df56c0ec3cbc8261ac57d333ad9c4d72a46ecdf8/go.mod#L432-L434

These shouldn't have been added.

Reply:
Done.

Reply:
We do need those seemingly redundant replaces actually, otherwise you get errors: https://github.com/straussb/amazon-cloudwatch-agent/actions/runs/8651520015/job/23722444316

I think it's because these are indirect dependencies, so doing `go mod edit -require` ends up producing these lines to set the version explicitly(?)